sdk/src/main/scala/com/horizen/certificatesubmitter/strategies/KeyRotationStrategy.scala

```diff
@@ -0,0 +1,95 @@
+package com.horizen.certificatesubmitter.strategies
+
+import akka.util.Timeout
+import com.horizen._
+import com.horizen.certificatesubmitter.CertificateSubmitter.SignaturesStatus
+import com.horizen.certificatesubmitter.dataproof.DataForProofGeneration
+import com.horizen.chain.{MainchainHeaderInfo, SidechainBlockInfo}
+import com.horizen.consensus.ConsensusEpochNumber
+import com.horizen.fork.ForkManager
+import com.horizen.params.NetworkParams
+import com.horizen.utils.{BytesUtils, TimeToEpochUtils}
+import scorex.util.ScorexLogging
+import sparkz.core.NodeViewHolder.CurrentView
+
+import java.io.File
+import scala.compat.java8.OptionConverters.RichOptionalGeneric
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
+
+abstract class KeyRotationStrategy(settings: SidechainSettings, params: NetworkParams) extends ScorexLogging{
```

Comment:
KeyRotationStrategy must be generic class, parameterized by DataForProofGeneration type.
So, `WithKeyRotationStrategy extends KeyRotationStrategy[DataForProofGenerationWithKeyRotation]`.
Same `for WithoutKeyRotationStrategy`